lib/sepa.js

```diff
@@ -243,46 +356,63 @@
      * @return      The DOM <PmtInf> Element.
      */
     toXML: function(doc) {
-      this.validate();
+      if (VALIDATIONS_ENABLED)
+        this.validate();
+
       var n = createXMLHelper(doc, true, false);
-      var o = createXMLHelper(doc, false, true);
+      //var o = createXMLHelper(doc, false, true);
       var r = createXMLHelper(doc, true, true);
-      var pmtInf = doc.createElementNS(doc.documentElement.namespaceURI, "PmtInf");
-
-      r(pmtInf, "PmtInfId", this.id);
-      r(pmtInf, "PmtMtd", this.method);
-      r(pmtInf, "BtchBookg", this.batchBooking);
-      r(pmtInf, "NbOfTxs", this.transactionCount);
-      r(pmtInf, "CtrlSum", this.controlSum.toFixed(2));
-
-      var pmtTpInf = n(pmtInf, "PmtTpInf");
-      r(pmtTpInf, "SvcLvl", "Cd", "SEPA");
-      r(pmtTpInf, "LclInstrm", "Cd", this.localInstrumentation);
-      r(pmtTpInf, "SeqTp", this.sequenceType);
-
-      r(pmtInf, "ReqdColltnDt", this.collectionDate.toISOString().substr(0, 10));
-      var creditor = n(pmtInf, "Cdtr");
-      r(creditor, "Nm", this.creditorName);
-      if (this.creditorStreet && this.creditorCity && this.creditorCountry) {
-        var pstl = n(creditor, "PstlAdr");
-        r(pstl, "Ctry", this.creditorCountry);
-        r(pstl, "AdrLine", this.creditorStreet);
-        r(pstl, "AdrLine", this.creditorCity);
+      var pmtInf = doc.createElementNS(doc.documentElement.namespaceURI, 'PmtInf');
+
+      r(pmtInf, 'PmtInfId', this.id);
+      r(pmtInf, 'PmtMtd', this.method);
+      // XML v3 formats, add grouping + batch booking nodes
+      if (getPainXMLVersion(this._painFormat) === 3) {
+        r(pmtInf, 'BtchBookg', this.batchBooking.toSring());
+        r(pmtInf, 'Grpg', this.grouping);
+        r(pmtInf, 'NbOfTxs', this.transactionCount);
+        r(pmtInf, 'CtrlSum', this.controlSum.toFixed(2));
+      }
+
+      var pmtTpInf = n(pmtInf, 'PmtTpInf');
+      r(pmtTpInf, 'SvcLvl', 'Cd', 'SEPA');
+      r(pmtTpInf, 'LclInstrm', 'Cd', this.localInstrumentation);
+
+      if (this.method === PaymentInfoTypes.DirectDebit) {
+        r(pmtTpInf, 'SeqTp', this.sequenceType);
+        r(pmtInf, 'ReqdColltnDt', this.collectionDate.toISOString().substr(0, 10));
+      }
+      else
+        r(pmtInf, 'ReqdExctnDt', this.requestedExecutionDate.toISOString().substr(0, 10));
```

Comment:
I prefer brackets even for one-line if/else blocks, I'd appreiate if you could fix this (and maybe adapt eslint, in case there is an option for that)